File: benchmarks/suite5/prime.zig
const bench = @import("bench.zig");
const clib = @cImport({
    @cInclude("stdio.h");
});

pub fn main() void {
    const limit: i64 = 100_000;
    var count: i64 = 0;
    var i: i64 = 2;
    while (i <= limit) : (i += 1) {
        var prime: i64 = 1;
        var j: i64 = 2;
        while (j * j <= i) : (j += 1) {
            if (@rem(i, j) == 0) {
                prime = 0;
                break;
            }
        }
        if (prime != 0) count += 1;
    }
    bench.printI64(count);
}

File: benchmarks/suite5/prime.c
#include <stdint.h>
#include <stdio.h>
#include "bench.h"

int main(void) {
  const int64_t limit = suite5_opaque_i64(100000);
  int64_t count = 0;
  for (int64_t i = 2; i <= limit; i++) {
    int64_t prime = 1;
    for (int64_t j = 2; j * j <= i; j++) {
      if (i % j == 0) {
        prime = 0;
        break;
      }
    }
    if (prime) {
      count++;
    }
  }
  suite5_print_i64(count);
  return 0;
}
